File: datechating-client/src/app/model/userspecparams.ts
import { User } from "./user";

export class Userspecparams{    
    pageSize = 8;
    pageIndex = 1;   
    gender : string;
    orderBy = 'lastActive';
    ageFrom = 18;
    ageTo = 100; 
    

    constructor(user : User){
        this.gender = user.gender === "female" ? "male" : "female";
    }
}

File: API/Model/UserSpecParams.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using API.Helpers;

namespace API.Model
{
    public class UserSpecParams : PaginationParams
    {
        

        public string  UserName { get; set; }
        public string  Gender { get; set; }

        public int AgeFrom { get; set; } = 18;

        public int AgeTo { get; set; } = 100;

        public string OrderBy { get; set; }
    }
}